test/foundry/OffchainAssetReceiptVaultFactory.t.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: CAL
+pragma solidity =0.8.17;
+
+import {VaultConfig} from "../../contracts/vault/receipt/ReceiptVault.sol";
+import {CreateOffchainAssetReceiptVaultFactory} from "../../contracts/test/CreateOffchainAssetReceiptVaultFactory.sol";
+import "forge-std/Test.sol";
+import {
+    OffchainAssetReceiptVault,
+    OffchainAssetVaultConfig,
+    OffchainAssetReceiptVaultConfig
+} from "../../contracts/vault/offchainAsset/OffchainAssetReceiptVault.sol";
+
+contract OffChainAssetReceiptVaultFactoryTest is Test, CreateOffchainAssetReceiptVaultFactory {
+    ///Test that OffchainAssetReceiptVaultFactory is created
+    function testOffchainAssetReceiptVaultFactoryConstuction() external {
+        //check address
+        assert(address(factory) != address(0));
+
+        //check codeSize
+        address factoryAddress = address(factory);
+        uint256 size;
+        assembly {
+            size := extcodesize(factoryAddress)
+        }
+        assertTrue(size > 0);
+    }
+
+    ///Test OffchainAssetReceiptVaultFactory child is created with correct properties
+    ///and that OffchainAssetReceiptVaultInitialized event is emitted
+    function testCreateChild(uint256 fuzzedKeyAlice, string memory assetName, string memory assetSymbol) external {
+        // Ensure the fuzzed key is within the valid range for secp256k1
+        fuzzedKeyAlice = bound(fuzzedKeyAlice, 1, SECP256K1_ORDER - 1);
+        address alice = vm.addr(fuzzedKeyAlice);
+
+        // VaultConfig to create child contract
+        VaultConfig memory vaultConfig = VaultConfig({asset: address(0), name: assetName, symbol: assetSymbol});
```

Comment:
we can fuzz the asset

Reply:
but shouldn't the asset be address zero?